tests/ui/unsafe-binders/moves.rs

```diff
@@ -0,0 +1,41 @@
+//@ known-bug: unknown
```

Comment:
This test *will* (after I add a new trait instead of using `Copy`) demonstrate that we reason about moves correctly with unsafe binders.